service/instance.py

```diff
@@ -1188,7 +1188,11 @@ def check_application_threshold(
     application = Application.objects.filter(
         versions__machines__instance_source__identifier=boot_source.identifier,
         versions__machines__instance_source__provider=core_identity.provider).distinct().get()
-    threshold = application.get_threshold()
+    try:
+        threshold = boot_source.current_source.application_version.get_threshold()
+    except:
+        threshold = application.latest_version.get_threshold()
+
     if not threshold:
         return
```

Comment:
The logic here should be:
try to retrieve threshold. If it fails -- There is no minimum threshold -- return.